edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/ModuleIdentityLifecycleManager.cs

```diff
@@ -44,7 +37,19 @@ public ModuleIdentityLifecycleManager(IIdentityManager identityManager, ModuleId
             }
         }
 
-        async Task<IImmutableDictionary<string, IModuleIdentity>> GetModuleIdentitiesAsync(Diff diff)
+        protected virtual async Task<IImmutableDictionary<string, IModuleIdentity>> GetModuleIdentitiesWorkAsync(ModuleSet desired, ModuleSet current)
```

Comment:
We should think about this design better. It's not clear what the goal is unless you read through both implementations. Perhaps create a virtual method that "manages" empty diff sets. Then, for Kube, you can have that be a noop (return true or something).

Just GetModuleIdentitiesWork is confusing because you're clearly abstracting away **one** behavior (empty diff handling) but that's not clear from the design of this virtual method.

Reply:
Also, scoping down the virtual behavior to something like empty diff handler will scope down the surface here. I could see someone in the future not realize what's happening and add generic behavior to one but not the other.

Reply:
Yes, that was @dmolokanov's suggestion, too.  I hated how it looked - essentially in one implementation, it's returning an empty dictionary, and in the other implementation, it's duplicating the rest of the function.  I'll try it again...